source/system/String.ooc

```diff
@@ -107,7 +107,11 @@ String: class {
 		match (this size) {
 			case 0 => this
 			case 1 => this toUpper()
-			case => this[0] toUpper() + this substring(1)
+			case =>
+				end := this substring(1)
+				result := this[0] toUpper() + end
+				end free()
+				result
```

Comment:
Immutability should affect only clients of the class, there is nothing wrong with bending the rules inside the implementation details. As long as it does not leak outside.